src/tests/Loader/ContextualReflection/ContextualReflection.cs

```diff
@@ -729,14 +729,15 @@ void TestActivatorCreateInstance(bool isolated)
         void TestDefineDynamicAssembly(bool collectibleContext, AssemblyBuilderAccess assemblyBuilderAccess)
         {
             AssemblyLoadContext assemblyLoadContext = collectibleContext ? new AssemblyLoadContext("DynamicAssembly Collectable context", true) : AssemblyLoadContext.Default;
-            AssemblyName dynamicAssemblyName = new AssemblyName("DynamicAssembly");
+            AssemblyBuilder assemblyBuilder;
+            AssemblyName assemblyName = new AssemblyName($"DynamicAssembly_{Guid.NewGuid():N}");
 
             using (assemblyLoadContext.EnterContextualReflection())
             {
-                AssemblyBuilder assemblyBuilder = AssemblyBuilder.DefineDynamicAssembly(dynamicAssemblyName, assemblyBuilderAccess);
+                assemblyBuilder = AssemblyBuilder.DefineDynamicAssembly(assemblyName, assemblyBuilderAccess);
             }
 
-            Assert.IsTrue(assemblyLoadContext.Assemblies.Any(a => AssemblyName.ReferenceMatchesDefinition(a.GetName(), dynamicAssemblyName)));
+            Assert.IsTrue(assemblyLoadContext == AssemblyLoadContext.GetLoadContext(assemblyBuilder));
```

Comment:
With the fix in CoreLib I would expect both approaches to work - the comparison of ALC as well as searching for the assembly in ALC.Assemblies. Is that true?

Reply:
Yes, provided that the name of the dynamic assembly is unique for each context (default and unloaded).

Reply:
Returned the search for dynamic assembly in ALC.Assemblies. The dynamic assembly name is unique for each ALC.